packages/browser/src/tracing/browserTracingIntegration.ts

```diff
@@ -282,6 +284,29 @@ export const browserTracingIntegration = ((_options: Partial<BrowserTracingOptio
         });
       });
 
+      // A trace should to stay the consistent over the entire time span of one route.
+      // Therefore, when the initial pageload or navigation transaction ends, we update the
+      // scope's propagation context to keep span-specific attributes like the `sampled` decision and
+      // the dynamic sampling context valid, even after the transaction has ended.
+      // This ensures that the trace data is consistent for the entire duration of the route.
+      client.on('spanEnd', span => {
+        const op = spanToJSON(span).op;
+        if (span !== getRootSpan(span) || (op !== 'navigation' && op !== 'pageload')) {
+          return;
+        }
+
+        const scope = getCurrentScope();
+        const oldPropagationContext = scope.getPropagationContext();
+
+        const newPropagationContext = {
+          ...oldPropagationContext,
+          sampled: oldPropagationContext.sampled !== undefined ? oldPropagationContext.sampled : spanIsSampled(span),
```

Comment:
l/m: Not 100% sure, but maybe it would be safer to pick this from the DSC, to ensure this is aligned? So something like this:

```js
const dsc = oldPropagationContext.dsc || getDynamicSamplingContextFromSpan(span);
const dscSampled = dsc.sampled === 'true' ? true : dsc.sampled === 'false' : false : undefined;
const sampled = oldPropagationContext.sampled !== undefined ? oldPropagationContext.sampled : dscSampled;
```

Reply:
Please do not use any values from the DSC in the SDK. This has potential to cause a lot of issues later on.

Reply:
Okay, so we quickly discussed this in person with @cleptric and settled on updating `propagationContext.sampled` from the propagation context and `spanIsSampled` instead of the DSC. The reason is that there are talks around removing `sampled` from the DSC again because it's (likely) unused in ingest/product. 

Both of the approaches currently lead to the same outcome, as demonstrated by the integration tests but it's safer in this case to read from `propagationContext` (i.e. `sentry-trace`) instead of DSC/`baggage`.